File: 522.longest-uncommon-subsequence-ii.js
/*
 * @lc app=leetcode id=522 lang=javascript
 *
 * [522] Longest Uncommon Subsequence II
 */
/**
 * @param {string[]} strs
 * @return {number}
 */
var findLUSlength = function(strs) {
	if (strs === null) return -1;
	strs.sort((a, b) => b.length - a.length);
	// get closure

	for (let i = 0; i < strs.length; i++) {
		let flag = true;
		for (let str of strs) {
			if (str.includes(strs[i])) {
				flag = false;
			}
		}
		if (flag) return strs[i].length;
	}
	return -1;
};

let b = findLUSlength([ 'aba', 'cdc', 'eae' ]);
console.log(b);
// var isSubequence(a, b){
//     return a.includes(b);

// }

File: 522.longest-uncommon-subsequence-ii.java
import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

/*
 * @lc app=leetcode id=522 lang=java
 *
 * [522] Longest Uncommon Subsequence II
 */

// @lc code=start
class Solution {
    public int findLUSlength(String[] strs) {
        Set<String> set = new HashSet<>();
        Collections.addAll(set, strs);
        int cnt = 0;
        for(String str: strs){
            if(str.length() > cnt)
                cnt = str.length();
            set.add(str);
        }
        //find whether is subcommon subseuqnce 
        
        if(set.size() == 1)
            return -1;
        return cnt;
        // set.addA(strs);
        
        
    }
}
// @lc code=end

